src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,71 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
-    </p>
-  </form>
-);
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList';
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoFilter: React.FC<Props> = ({ todos }) => {
+  const [query, setQuery] = useState('');
+  const [filtered, setFilter] = useState('');
+
+  let filteredTodos = todos.filter((todo: { title: string }) => todo
+    .title.toLowerCase().includes(query.toLowerCase()));
+
+  if (filtered === 'active') {
+    filteredTodos = filteredTodos.filter((todo) => !todo.completed);
+  } else if (filtered === 'completed') {
+    filteredTodos = filteredTodos.filter((todo) => todo.completed);
+  }
+
+  return (
+    <>
+      <form className="field has-addons">
+        <p className="control">
+          <span className="select">
+            <select
+              data-cy="statusSelect"
+              onChange={(event) => setFilter(event.target.value)}
```

Comment:
It's better to create **handleChange** method and use it here

Reply:
It is not big function. I don't think that i need to make new memory cell for this

Reply:
It's more about the clarity of the code. In the future, it will be better to edit code in a separate function than in the event handler, ofc it's not necessary.

Reply:
Not critical here, but @DenysKleimenov is right, better to create methods in constants, then we'll be able to wrap it in useCallback and will reduce amount of logic written in JSX